apps/settings/lib/Hooks.php

```diff
@@ -215,6 +215,33 @@ public function onChangeEmail(IUser $user, $oldMailAddress) {
 		}
 	}
 
+	public function onChangeDisplayName(IUser $user, string $oldDisplayName) {
+		if ($oldDisplayName === $user->getDisplayName() ||
+			$user->getLastLogin() === 0) {
+			// display name didn't really change or user didn't login,
+			// so don't create activities.
+			return;
+		}
+
+		$event = $this->activityManager->generateEvent();
+		$event->setApp('settings')
+			->setType('personal_settings')
+			->setAffectedUser($user->getUID());
+
+		$actor = $this->userSession->getUser();
+		if ($actor instanceof IUser) {
+			$subject = Provider::DISPLAY_NAME_CHANGED_SELF;
+			if ($actor->getUID() !== $user->getUID()) {
+				$subject = Provider::DISPLAY_NAME_CHANGED_BY;
+			}
+			$event->setAuthor($actor->getUID())
+				->setSubject($subject, [$actor->getUID()]);
```

Comment:
```suggestion
				->setSubject($subject, ['actor' => $actor->getUID()]);
```

It proofed to be better to name parameters. (Also needs adjusting in the provider)

Reply:
Fixed.